src/core/lib/security/security_connector/tls/tls_security_connector.cc

```diff
@@ -318,9 +318,12 @@ int TlsChannelSecurityConnector::cmp(
 bool TlsChannelSecurityConnector::check_call_host(
     absl::string_view host, grpc_auth_context* auth_context,
     grpc_closure* /*on_call_host_checked*/, grpc_error** error) {
-  return grpc_ssl_check_call_host(host, target_name_.c_str(),
-                                  overridden_target_name_.c_str(), auth_context,
-                                  error);
+  if (options_->server_verification_option() == GRPC_TLS_SERVER_VERIFICATION) {
```

Comment:
One nit: can we make it 
```
if (options_->server_verification_option() == XXX || options_->server_verification_option() == XXX) return true;
return grpc_ssl_check_call_host(...);
```
That way, if we later on add a fourth field in verification option and forgot to update the code here, at least the connection would fail instead of allowing by default.

Reply:
Done